src/index.cpp

```diff
@@ -68,7 +68,12 @@ Index::Index(const std::string& name, const uint32_t collection_id, const Store*
         }
 
         if(a_field.num_dim > 0) {
-            auto hnsw_index = new hnsw_index_t(a_field.num_dim, 16, a_field.vec_dist, a_field.hnsw_params["M"].get<uint32_t>(), a_field.hnsw_params["ef_construction"].get<uint32_t>());
+            auto num_dim = a_field.num_dim;
+            if(a_field.is_bool()) {
+                num_dim = num_dim / (8 * sizeof(float)) + (num_dim % (8 * sizeof(float)) == 0 ? 0 : 1);
```

Comment:
Add a comment that we are converting binary dimensions to float dimensions since we will store the values as float.